src/tezos_rpc/fetch_block_header.ml

```diff
@@ -0,0 +1,23 @@
+open Tezos
+open Http
+
+type response = Block_header.t
+
+let path ~chain ~block_hash =
+  (* TODO: I don't like this Format.sprintf *)
+  Format.sprintf "/chains/%s/blocks/%s/header" chain block_hash
+
+let execute ~node_uri ~chain ~block_hash =
+  let chain =
```

Comment:
In Haskell, this would lead to `shadowing` how is it possible in OCaml? Just because of the `~`?
At first sight, I got confused, is there any OCaml (or Deku) convention to avoid this kind of misleading?
I am used to do something like
```ocaml
  let chain' =
    match chain with
    | Some chain -> Chain_id.to_string chain
    | None -> "main" in
```
Moreover, I am really curious to understand how:
```ocaml
let path = path ~chain ~block_hash in
```
is able to get this `chain` and not the `~chain` provided line 10.

Would you mind help me figure out how this is working? 😄

Reply:
OCaml values are not recursive by default, unlike Haskell(and most languages), so you can freely shadow variables.

It's the same as the following in Haskell.
```haskell
let x = 1 in
(\x -> print x) (x + 1)
```

Types are recursive by default but you can do non recursive types using `type nonrec t = t`.

Reply:
Thank you for the explanation 🙏!